### ISSUE TITLE
where can i find the pretrained model weights ? 

### ISSUE THREAD

**Issue by amrahsmaytas** [trinanjan12/Image-Based-Virtual-Try-on-Network-from-Unpaired-Data#2]
HI @trinanjan12 ,

thanks for the paper , really enjoyed it reading, 
I am trying to implement your inference code in google colab (would happy to contribute a pr, once completed), 
but there are few missing elements in your code (test.ipynb) i.e the pretrained model (./checkpoints/zalando_shape/latest_net_G.pth not exists yet!)

could you please be kind to 

1. share the pretrained model/weights ?
2. guide me on the steps to run the inference on custom images?
3. Till when, we  can expect the whole code to be released with proper documentation?
4. Any help,you need in contribution of the project from open source community, we are happy to help and contribute?

Thanks,in advance,
Satyam sharma.

**Comment by trinanjan12** [comment]
Hey @amrahsmaytas .I'm planning to make a better documentation in future. I'm busy with lot of things currently. I might do it near future. Regarding Google colab, i don't immediate plans for for rolling out right now.
Check out deepfashoon tryon or cpviton plus if you need immediately.

**Comment by amrahsmaytas** [comment]
Tried checking them out @trinanjan12 , but, i am facing some issue with them , i did mentioned you , regarding the steps, please go through the issue in the deepfashiontryon repo and please help me with the steps!

Thanks for the quick reply 

P.s: Could you please answer the above 1,2 questions ?

**Comment by trinanjan12** [comment]
@amrahsmaytas I am not much aware of deepdashion tryon and cpviton plus, and the other repo down to the last detail virtual tryon provides test code that worked for me. Try running the inference code, errors could be mostly because of pytorch small issues.Issues will help you! Good luck :)

**Comment by amrahsmaytas** [comment]
Sorry, i am talking about the same (last detail virtual try on), and i have gone through your discussion regarding test images at https://github.com/JDAI-CV/Down-to-the-Last-Detail-Virtual-Try-on-with-Detail-Carving/issues/5 but , could you please share me the code/ clear steps (i know, you said, you dont have the code, but can you point me to any repo, which is an working version of it,also checking your version of the last detail virtual try on)

Thanks again

**Comment by trinanjan12** [comment]
@amrahsmaytas  I haven't worked on it for a while. So I don't think I will be of much help. My repo is almost the same version of them as I change the code to try experimenting with densepose, and I don't remember getting better results.

**Comment by amrahsmaytas** [comment]
Bro, I guess , you have successfully implemented the last detail virtual try on test images, didn't you? 

If yes, please share the steps/code, I will look into it and will manage inference on test images! 

Thanks 🙂

**Comment by trinanjan12** [comment]
I'll try to checkout next week

**Comment by amrahsmaytas** [comment]
> I'll try to checkout next week

Thanks, will be waiting for your reply!

**Comment by trinanjan12** [comment]
@amrahsmaytas I just checked out the original repo https://github.com/JDAI-CV/Down-to-the-Last-Detail-Virtual-Try-on-with-Detail-Carving. The test code works fine with one modification in the geometric_matching_multi_gpu file. 
To test it on a custom dataset you need to prepare the following
1. a product cloth image(RGB)
2. the binary mask of the clothing images (it's easier for images with white background otherwise try grab cut. a word of caution the I haven't found any segmentation models that do product image segmentation because most of them are trained on full-body images)
3. Human model image for targets(RGB)
4. The corresponding segmentation map of the model image (I used chip_png first but it's tedious to install that. better use graphology. use an appropriate model of graphology which gives 20 channel output )
5. Keypoints of the model image(again if you are able to install oppose than fine otherwise use any models from detection keypoints)

follow this issue https://github.com/JDAI-CV/Down-to-the-Last-Detail-Virtual-Try-on-with-Detail-Carving/issues/5 for you reference. The colab notebook I shared should be used for reference only. The code for inference given in the original repo is correct, so you have to follow the data preparation EXACTLY. 
A couple of things to remember, use proper JSON str for the pose file(refer to the ones that have been shared), use proper naming convention as the names are directly used in the demo.py file. example source_pose = a['people'][0]['pose_keypoints_2d'] line number 101 demo.py file. nowhere if you don't follow the exact JSON str this won't work.

Good Luck!

**Comment by amrahsmaytas** [comment]
Thanks a lot, for the clear steps, I would try them as mentioned, and hope it works! 

Thanks again, for being so responsive, @trinanjan12

**Comment by amrahsmaytas** [comment]
HI @trinanjan12 

1. whats that one modification in geometric_matching_multi_gpu to run it successfully?

2. The prepare_data.ipynb as mentioned at https://colab.research.google.com/drive/1IM84ldwD36XTEPN90PVenzj2R_Q1lUad#scrollTo=HgqxMFadPRLV is not running smoothly ,if possible to can you update  the colab with a working version, which you can find in /media/trinanjan/BACKUP/Trinanjan/current_project/virtual_try_on/

Thanks
